plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -354,10 +356,11 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
         List<SchemaTableName> tablesList = schemaName.map(Collections::singletonList)
                 .orElseGet(metastore::getAllDatabases)
                 .stream()
-                .flatMap(schema -> metastore.getTablesWithParameter(schema, TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE).stream()
-                        .map(table -> new SchemaTableName(schema, table))
-                        .collect(toList())
-                        .stream())
+                .flatMap(schema -> Stream.concat(
```

Comment:
You gonna have duplicates when using HMS with case insensitive backend. 
Add `.distinct()` to the steam _within_ `flatMap`.